mkosi/__init__.py

```diff
@@ -726,35 +715,37 @@ def make_tar(state: MkosiState) -> None:
         run(cmd)
 
 
-def find_files(root: Path) -> Iterator[Path]:
-    """Generate a list of all filepaths relative to @root"""
-    yield from scandir_recursive(root,
+def find_files(dir: Path, root: Path) -> Iterator[Path]:
+    """Generate a list of all filepaths in directory @dir relative to @root"""
+    yield from scandir_recursive(dir,
                                  lambda entry: Path(entry.path).relative_to(root))
 
 
-def make_cpio(state: MkosiState) -> None:
+def make_initrd(state: MkosiState) -> None:
     if state.do_run_build_script:
         return
     if state.config.output_format != OutputFormat.cpio:
         return
     if state.for_cache:
         return
 
-    with complete_step("Creating archive…"), open(state.staging / state.config.output.name, "wb") as f:
-        files = find_files(state.root)
+    make_cpio(state.root, find_files(state.root, state.root), state.staging / state.config.output.name)
+
+
+def make_cpio(root: Path, files: Iterator[Path], output: Path) -> None:
+    with complete_step("Creating archive…"):
         cmd: list[PathString] = [
-            "cpio", "-o", "--reproducible", "--null", "-H", "newc", "--quiet", "-D", state.root
+            "cpio", "-o", "--reproducible", "--null", "-H", "newc", "--quiet", "-D", root, "-O", output
         ]
 
-        with spawn(cmd, stdin=subprocess.PIPE, stdout=f) as cpio:
+        with spawn(cmd, stdin=subprocess.PIPE, text=True) as cpio:
             #  https://github.com/python/mypy/issues/10583
             assert cpio.stdin is not None
 
             for file in files:
-                cpio.stdin.write(os.fspath(file).encode("utf8") + b"\0")
+                cpio.stdin.write(os.fspath(file))
+                cpio.stdin.write("\0")
```

Comment:
This change seems not necessary. Why decode and reencode the bytes when we're just passing them through?

Reply:
Ah, right... Forgot about that.

Reply:
That comment was actually meant for the other thread.